source/mixin/network_mixin.py

```diff
@@ -0,0 +1,40 @@
+from PyQt4.QtCore import *
+from PyQt4.QtNetwork import *
+
+
+class NetworkMixin(QNetworkAccessManager):
+    request_url = str
+    result_connection = ""
+
+    @pyqtSlot()
+    def connectionFinished(self):
+        print(self.result_connection)
+
+    @pyqtSlot()
+    def connectionReadData(self):
+        self.result_connection += self.reply.readAll()
+
+    def connect(self):
+        print "connect it"
+        QObject.connect(
```

Comment:
'QObject.connect' is the old way to connect signals. You should do something like : https://github.com/inasafe/inasafe/blob/develop/safe/gui/tools/osm_downloader_dialog.py#L99
'python_object.signal_name.connect(function)'